### PR TITLE
feat: UP-7465 - sync, test app - update TS-SDK

### PR DESCRIPTION
**Pull request by ramyareye**
Update app sdk version to latest and do QA 
update functionalities and upgrade expo

### CODE REVIEW

File: src/components/AddComment/index.tsx [hunk 1]
```diff
@@ -116,6 +120,10 @@ const AddComment: VFC<AddCommentType> = ({
         if (data) {
           setText('');
           onRefresh();
+
+          if (isReply !== '') {
+            navigation.navigate('Comments', { postId, parentId: isReply });
```
Comment (kaungmyatlwin) [inline src/components/AddComment/index.tsx:125]: I think it's better to extract `'Comments'` screen to a constant file, import and reuse it. This way we wouldn't be confused of naming in the future.

File: src/utils/getErrorMessage.ts [hunk 2]
```diff
@@ -1,6 +1,6 @@
-import { AxiosError } from 'axios';
+// import { AxiosError } from 'axios';
```
Comment (kaungmyatlwin) [inline src/utils/getErrorMessage.ts:1]:
```suggestion
```

File: src/components/CommunityItem/index.tsx [hunk 3]
```diff
@@ -122,21 +137,27 @@ const CommunityItem: VFC<{ community: Amity.Community } & CommunityItemProps> =
...
+
+      <Card.Actions
+        key={`Actions_${community?.membersCount}_${community?.postsCount}`}
+        style={styles.footer}
+      >
         <View style={styles.footerLeft}>
           <View style={styles.subtitleRow}>
             <MaterialCommunityIcons size={18} name="account-group" color={textColor} />
-            <Text>{community?.membersCount}</Text>
+            {!!community && <Text>{String(community?.membersCount ?? 0)}</Text>}
```
Comment (kaungmyatlwin) [inline src/components/CommunityItem/index.tsx:152]: Curious here. Why should we need to wrap this in `String` constructor? Doesn't it render the int types too?

File: src/components/PostItem/index.tsx [hunk 4]
```diff
@@ -80,11 +85,22 @@ const PostItem: VFC<{ post: Amity.Post } & PostItemProps> = ({
 
   useEffect(() => {
     return observePost(postId, updatedPost => {
+      // console.log('observePost', { updatedPost });
```
Comment (kaungmyatlwin) [inline src/components/PostItem/index.tsx:88]:
```suggestion
```

File: src/components/PostItem/index.tsx [hunk 5]
```diff
@@ -157,12 +173,17 @@ const PostItem: VFC<{ post: Amity.Post } & PostItemProps> = ({
     alertConfirmation(() => {
       setOpenMenu(false);
 
-      runQuery(createQuery(deletePost, postId), ({ data: postData, error }) => {
+      // const isPermanentDelete = !!isDeleted;
+      // console.log(1);
```
Comment (kaungmyatlwin) [inline src/components/PostItem/index.tsx:177]:
```suggestion
```

File: src/components/PostItem/index.tsx [hunk 6]
```diff
@@ -157,12 +173,17 @@ const PostItem: VFC<{ post: Amity.Post } & PostItemProps> = ({
...
+
+      runQuery(createQuery(deletePost, postId, true), ({ data: postData, error }) => {
+        // console.log(2, { postData, error });
         if (postData) {
           if (!onPress) {
             navigation.goBack();
           }
         } else if (error) {
+          // console.log(3, { error });
```
Comment (kaungmyatlwin) [inline src/components/PostItem/index.tsx:186]:
```suggestion
```

File: src/components/PostItem/index.tsx [hunk 7]
```diff
@@ -157,12 +173,17 @@ const PostItem: VFC<{ post: Amity.Post } & PostItemProps> = ({
     alertConfirmation(() => {
       setOpenMenu(false);
 
-      runQuery(createQuery(deletePost, postId), ({ data: postData, error }) => {
+      // const isPermanentDelete = !!isDeleted;
+      // console.log(1);
+
+      runQuery(createQuery(deletePost, postId, true), ({ data: postData, error }) => {
+        // console.log(2, { postData, error });
```
Comment (kaungmyatlwin) [inline src/components/PostItem/index.tsx:180]:
```suggestion
```

File: src/screens/Communities/index.tsx [hunk 8]
```diff
@@ -1,33 +1,38 @@
...
+const QUERY_LIMIT = 10;
 
 const CommunitiesScreen: VFC = () => {
   const [isEditId, setIsEditId] = useState('');
+  const [isRefreshing, setIsRefreshing] = useState(false);
   const [showAddCommunity, setShowAddCommunity] = useState(false);
   const [isDeleted] = React.useState<Amity.Community['isDeleted']>(false);
-  const [membership] = useState<'member' | 'notMember' | 'all'>('member');
-  const [loading, setLoading] = useState<LoadingState>(LoadingState.NOT_LOADING);
+  const [membership] = useState<'member' | 'notMember' | 'all'>('all');
```
Comment (kaungmyatlwin) [inline src/screens/Communities/index.tsx:22]: AFAIK there's also a data type called `banned` for communities.

File: src/screens/Communities/index.tsx [hunk 9]
```diff
@@ -91,7 +92,6 @@ const CommunitiesScreen: VFC = () => {
 
   const handleLoadMore = () => {
     if (nextPage) {
-      setLoading(LoadingState.IS_LOADING_MORE);
       onQueryCommunities({ page: nextPage });
     }
   };
```
Comment (kaungmyatlwin) [inline src/screens/Communities/index.tsx:97]: We can extract this logic to a hook `usePagination`? Check it out here: https://github.com/EkoCommunications/AmityConsoleWeb/blob/develop/src/hooks/usePagination.js

File: src/screens/Community/index.tsx [hunk 10]
```diff
@@ -54,6 +54,13 @@ const Community: VFC = () => {
     });
   }, [communityId, navigation]);
 
+  const addUser = useCallback(async () => {
+    // const query = createQuery(removeCommunityMembers, communityId, ['Noor']);
+    // runQuery(query, data => {
+    //   console.log({ data });
+    // });
+  }, []);
```
Comment (kaungmyatlwin) [inline src/screens/Community/index.tsx:62]:
```suggestion
```

Redundant callback function?

File: src/screens/Post/index.tsx [hunk 11]
```diff
@@ -69,8 +69,8 @@ const PostScreen: VFC = () => {
   }, []);
 
   const onEditPost = useCallback(() => {
-    setIsEditId(postId);
-  }, [postId]);
+    // setIsEditId(postId);
+  }, []);
```
Comment (kaungmyatlwin) [inline src/screens/Post/index.tsx:73]: Redundant callback here too.

File: src/screens/Users/index.tsx [hunk 12]
```diff
@@ -56,15 +59,15 @@ const UserListScreen: VFC = () => {
...
-        setUsers(prevUsers => ({ ...prevUsers, ...data }));
-
-        // @ts-ignore
-        setMetadata(metadata);
+      // let uniqueObjArray = [...new Map(objArray.map((item) => [item["id"], item])).values()];
+      runQuery(createQuery(queryUsers, queryData), ({ data, ...metadata }) => {
+        if (data) {
+          setUsers(prevUsers => (reset ? data : [...prevUsers, ...data]));
+        }
+        // console.log({ data, metadata });
```
Comment (kaungmyatlwin) [inline src/screens/Users/index.tsx:67]:
```suggestion
```

File: src/screens/Users/index.tsx [hunk 13]
```diff
@@ -109,16 +112,17 @@ const UserListScreen: VFC = () => {
   );
 
   const errorText = getErrorMessage(error);
-  const data = Object.values(users).sort(sortByLastCreated);
+  const data = users.sort(sortByLastCreated);
+
+  // console.log({ users, data });
```
Comment (kaungmyatlwin) [inline src/screens/Users/index.tsx:117]:
```suggestion
```

File: src/screens/Users/index.tsx [hunk 14]
```diff
@@ -109,16 +112,17 @@ const UserListScreen: VFC = () => {
   );
 
   const errorText = getErrorMessage(error);
-  const data = Object.values(users).sort(sortByLastCreated);
+  const data = users.sort(sortByLastCreated);
```
Comment (kaungmyatlwin) [inline src/screens/Users/index.tsx:115]: I believe you can memoize this to reduce re-renders.

```suggestion
  const data = useMemo(() => users.sort(sortByLastCreated), [users, sortByLastCreated]);
```

**Review by Va1tra — APPROVED** [review]

**Pull request merged**